src/v/kafka/protocol/schemata/generator.py

```diff
@@ -309,6 +315,63 @@
     fetch_record_set=("batch_reader", None, "read_nullable_batch_reader()"),
 )
 
+# Declare some fields as sensitive. Utmost care should be taken to ensure the
+# contents of these fields are never made available over unsecure channels
+# (sent over an unencrypted connection, printed in logs, etc.).
+sensitive_map = {
+    "AlterConfigsRequestData": {
+        "Resources": {
+            # NOTE: this doesn't take into account whether the altered config
+            # is sensitive. Since there is no IsSensitive field in the
+            # AlterableConfig, assume all configs are sensitive.
+            "Configs": {
+                "Value": True,
+            },
+        },
+    },
+    "DescribeConfigsResponseData": {
+        "Results": {
+            # TODO: this doesn't take into account whether the altered config
+            # is sensitive. There is an IsSensitive field in the
+            # DescribeConfigsResourceResult, that we may be able to leverage.
+            # For now, treat all configs as sensitive.
+            "Configs": {
+                "Value": True,
+            },
+        },
+    },
+    "FetchResponseData": {
+        "Topics": {
+            "Partitions": {
+                "Records": True,
+            },
+        },
+    },
+    "IncrementalAlterConfigsRequestData": {
+        "Resources": {
+            # NOTE: this doesn't take into account whether the altered config
+            # is sensitive. Since there is no IsSensitive field in the
+            # AlterableConfig, assume all configs are sensitive.
+            "Configs": {
+                "Value": True,
+            },
+        },
+    },
+    "ProduceRequestData": {
+        "Topics": {
+            "Partitions": {
+                "Records": True,
```

Comment:
this is probably ok? afaict it only prints the size of the message which might be useful for debugging. however i think leaving a note in those operator<< overloads like you did for the scram_credentials etc would be robust enough. same for the Records in fetch?

Reply:
Ah, good catch. I assumed bytes types were logged as sstrings. Added comments to the corresponding operator<<s.